resources/indexer.py

```diff
@@ -18,290 +19,287 @@
+
+
+class BasicFileMetadata:
+    """Metadata for basic files"""
+    file = None
+    site = ""
+
+    def __init__(self, file, site):
+        self.file = file
+        self.site = site
+
+    def generate(self):
+        """Gather the file's metadata."""
+        metadata = dict()
+        metadata['logical_name'] = self.file.path
+        metadata['checksum'] = {'sha512': self.sha512sum()}
+        metadata['file_size'] = self.file.stat().st_size
+        metadata['locations'] = [{'site': self.site, 'path': self.file.path}]
+        metadata['create_date'] = date.fromtimestamp(os.path.getctime(self.file.path)).isoformat()
+        return metadata
+
+    def sha512sum(self):
+        """Return the SHA512 checksum of the file given by path."""
+        bufsize = 4194304
+        sha = hashlib.new('sha512')
+        with open(self.file.path, 'rb', buffering=0) as file:
+            line = file.read(bufsize)
+            while line:
+                sha.update(line)
+                line = file.read(bufsize)
+        return sha.hexdigest()
+
+
+class I3FileMetadata(BasicFileMetadata):
+    """Metadata for i3 files"""
+    processing_level = None
+    season_year = None
+    run = 0
     subrun = 0
+    part = 0
+    meta_xml = dict()
+
+    def __init__(self, file, site, processing_level):
+        super().__init__(file, site)
+        self.processing_level = processing_level
+        self._parse_filepath()
+
+    def generate(self):
+        """Gather the file's metadata."""
+        metadata = super().generate()
+
+        start_dt, end_dt, create_date, software = self._parse_xml()
+        data_type = self._get_data_type()
+        first_event, last_event, event_count, status = self._get_events_data()
+
+        metadata['create_date'] = create_date  # Override BasicFileMetadata's value
+        metadata['data_type'] = data_type
+        metadata['processing_level'] = self.processing_level
+        metadata['content_status'] = status
+        metadata['software'] = software
+
+        if data_type == "real":
+            metadata['run'] = {
+                'run_number': self.run,
+                'subrun_number': self.subrun,
+                'part_number': self.part,
+                'start_datetime': start_dt,
+                'end_datetime': end_dt,
+                'first_event': first_event,
+                'last_event': last_event,
+                'event_count': event_count
+            }
+        return metadata
+
+    def _parse_filepath(self):
+        """Set the year, run, subrun, and part from the file name."""
+        pass
+
+    @staticmethod
+    def parse_run_number(file):
+        """Return run number from filename."""
+        # Ex. Level2_IC86.2017_data_Run00130484_0101_71_375_GCD.i3.zst
+        # Ex: Level2_IC86.2017_data_Run00130567_Subrun00000000_00000280.i3.zst
+        # Ex: Run00125791_GapsTxt.tar
+        r = file.name.split('Run')[1]
+        run = int(r.split('_')[0])
+        return run
```

Comment:
You've got a `file` attribute of your base class object (BasicFileMetadata), so it would seem natural to use it in your subclass (I3Metadata) instead of making this method static and passing it as an argument.

If you want to keep the static method, I'd refactor it out of the class and keep it as a stand-alone function.